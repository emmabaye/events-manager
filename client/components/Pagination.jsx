import React, { Component } from 'react';
import { Link } from 'react-router-dom';

/**
 * React component to display all pagination buttons.
 */
export default class Pagination extends Component {
  /**
   * React's componentDidMount life cycle method
   * runs after component has been mounted.
   * Makes style changes after component has been mounted
   *
   * @return {undefined}
   */
  componentDidMount() {

  }

  handleGetItems = (page) => {
    this.props.dispatch(this.props.getItems(page));
  }

  /**
   * React's method to render react component.
   *
   * @return {object}
   */
  render() {
    return (
      <nav aria-label="..."
        style={{ display: "flex", flexDirection: "row", justifyContent: "center" }}>
        <ul className="pagination">
          <li className={`page-item gg ${(this.props.currentPage === this.props.firstPage) ? "disabled" : ""}`}
            onClick={() => this.handleGetItems(this.props.previousPage)}>
            <Link className="page-link" tabIndex="-1" to="#">Previous</Link>
          </li>
          { this.props.previousPage - 1 > this.props.firstPage &&
           <li className="page-item aa" onClick={() => this.handleGetItems(this.props.previousPage - 1)}>
             <Link className="page-link" to="#">{this.props.previousPage - 1}</Link>
           </li>
          }
          { this.props.previousPage !== "" &&
          <li className="page-item bb" onClick={() => this.handleGetItems(this.props.previousPage)}>
            <Link className="page-link" to="#">{this.props.previousPage}</Link>
          </li>
          }
          <li className="page-item cc active" onClick={() => this.handleGetItems(this.props.currentPage)}>
            <Link className="page-link" to="#">{this.props.currentPage} <span className="sr-only">(current)</span></Link>
          </li>
          { this.props.currentPage !== this.props.lastPage &&
            <li className="page-item dd" onClick={() => this.handleGetItems(this.props.nextPage)}>
              <Link className="page-link" to="#">{this.props.nextPage}</Link>
            </li>
          }
          { this.props.currentPage + 1 < this.props.lastPage &&
            <li className="page-item ee" onClick={() => this.handleGetItems(this.props.nextPage + 1)}>
              <Link className="page-link" to="#">{this.props.nextPage + 1}</Link>
            </li>
          }
          <li className={`page-item ff  ${(this.props.currentPage === this.props.lastPage) ? "disabled" : ""}`}
            onClick={() => this.handleGetItems(this.props.nextPage)}>
            <Link className="page-link" to="#">Next</Link>
          </li>
        </ul>
      </nav>
    );
  }
}

