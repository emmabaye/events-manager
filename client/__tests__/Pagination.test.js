import sinon from 'sinon';
import Pagination from '../components/Pagination.jsx';

const props = {
  firstPage: 1,
  currentPage: 4,
  previousPage: 3,
  nextPage: 5,
  lastPage: 6,
  dispatch: () => {},
  getItems: () => {}
};

describe('Pagination component should render self', () => {
  it('it should render for Pagination component', () => {
    const { wrapper } = shallow(<Pagination {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('Pagination events', () => {
  it('it should handle click events for list item with class bb', () => {
    const wrapper = shallow(<Pagination { ...props}/>);
    const spy = sinon.spy(wrapper.instance(), 'handleGetItems');
    const listItem = wrapper.find('.page-item .bb');
    listItem.simulate('click', {});
    expect(spy.calledOnce).toBeTruthy();
  });
});

describe('Pagination events', () => {
  it('it should handle click events for list item with class cc', () => {
    const wrapper = shallow(<Pagination { ...props}/>);
    const spy = sinon.spy(wrapper.instance(), 'handleGetItems');
    const listItem = wrapper.find('.page-item .cc');
    listItem.simulate('click', {});
    expect(spy.calledOnce).toBeTruthy();
  });

  it('it should handle click events for list item with class dd', () => {
    const wrapper = shallow(<Pagination { ...props}/>);
    const spy = sinon.spy(wrapper.instance(), 'handleGetItems');
    const listItem = wrapper.find('.page-item .dd');
    listItem.simulate('click', {});
    expect(spy.calledOnce).toBeTruthy();
  });

  it('it should handle click events for list item with class ee', () => {
    const wrapper = shallow(<Pagination { ...props}/>);
    const spy = sinon.spy(wrapper.instance(), 'handleGetItems');
    const listItem = wrapper.find('.page-item .ee');
    listItem.simulate('click', {});
    expect(spy.calledOnce).toBeTruthy();
  });

  it('it should handle click events for list with class ff', () => {
    const wrapper = shallow(<Pagination { ...props}/>);
    const spy = sinon.spy(wrapper.instance(), 'handleGetItems');
    const listItem = wrapper.find('.page-item .ff');
    listItem.simulate('click', {});
    expect(spy.calledOnce).toBeTruthy();
  });

  it('it should handle click events for list with class gg', () => {
    const wrapper = shallow(<Pagination { ...props}/>);
    const spy = sinon.spy(wrapper.instance(), 'handleGetItems');
    const listItem = wrapper.find('.page-item .gg');
    listItem.simulate('click', {});
    expect(spy.calledOnce).toBeTruthy();
  });

  it('it should handle click events for list with class aa', () => {
    const wrapper = shallow(<Pagination {...props}/>);
    const spy = sinon.spy(wrapper.instance(), 'handleGetItems');
    const listItem = wrapper.find('.page-item .aa');
    listItem.simulate('click', {});
    expect(spy.calledOnce).toBeTruthy();
  });
});
