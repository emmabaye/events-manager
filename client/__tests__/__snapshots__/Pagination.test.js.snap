// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination component should render self it should render for Pagination component 1`] = `undefined`;
